src/main.rs

```diff
@@ -18,8 +19,15 @@ pub struct LinePart {
     tab_index: Option<usize>,
 }
 
+#[derive(Debug, Default, Clone)]
+struct WatchStatus {
+    viewed: bool,
```

Comment:
leftover from previous vector-based implementation, I think it can be removed and the code simplified since the fact that a tab is in the hashmap implies it's not been viewed 🤔